### ISSUE TITLE
Data missing from JSON, when module run multiple times.

### ISSUE THREAD

**Issue by fgvieira** [ewels/MultiQC#2203]
### Description of bug

I am trying to plot the `FastQC` module multiple times (like [here](https://multiqc.info/docs/reports/customisation/#running-modules-multiple-times)) but, even though all looks good in the `html` report, only the latest data is stored in the `json`. That is, in the example above, only the `FastQC (trimmed)` data is stored in the `json`; is this intended?

Would it be possible to save both? Maybe by renaming `fastqc_sequence_length_distribution_plot` to `fastqc_trimmed-sequence_length_distribution_plot` and `fastqc_raw-sequence_length_distribution_plot` (like in the rest of the report)?

### MultiQC Error log

```console
No error reported.
```

### Before submitting

- [X] I have read the [troubleshooting documentation](https://multiqc.info/docs/usage/troubleshooting/).
- [X] I am using the latest release of MultiQC.
- [X] I have included a full MultiQC log, not truncated.
- [X] I have attached an input file (**.zip** if necessary) that triggers the error.

**Comment by vladsavelyev** [comment]
Thank you @fgvieira for reporting an issue - we definitely should save both, sounds like a bug. I'll take a look now

**Comment by fgvieira** [comment]
Which version are you using?
Because it is not on the JSON file [multiqc_data.json](https://github.com/ewels/MultiQC/files/13532051/multiqc_data.json)

**Comment by vladsavelyev** [comment]
I tried both `1.18 (af792aa)` (the latest stable release installed with `pip install multiqc==1.18`) as well as master (`1.19.dev0 (af792aa)`.

I wonder if it has to do with the input data? I can see e.g. both `fastqc_per_base_sequence_quality_plot` and `fastqc_per_base_sequence_quality_plot-1` in your JSON. And `sequence_length_distribution-1` might be missing because the `sequence_length_distribution` data was missing in the FastQC output for the `FastQC (raw)` samples?

Wonder if you run MultiQC in verbose mode with `-v`, it will print `sequence_length_distribution not found in FastQC reports` into the stderr?

If you could attach the FastQC reports are you running on, along with the `multiqc_config.yaml`, that would be really helpful for me to reproduce the issue!

**Comment by fgvieira** [comment]
I did not see anything out of the ordinary, but here is some example data:

[fastqc_raw.zip](https://github.com/ewels/MultiQC/files/13532292/fastqc_raw.zip)
[fastqc_trim.zip](https://github.com/ewels/MultiQC/files/13532295/fastqc_trim.zip)
[multiqc_config.zip](https://github.com/ewels/MultiQC/files/13532297/multiqc_config.zip)

PS - had to zip them separately due to GitHub size limitations.

**Comment by vladsavelyev** [comment]
If you look at the HTML report, that plot is not built for the "raw" inputs as I guess the logic was that it doesn't make sense when all reads are of the same length:

<img width="908" alt="Screenshot 2023-12-01 at 23 29 31" src="https://github.com/ewels/MultiQC/assets/1575412/01c2708d-8651-41aa-9ea7-dad606648630">

**Comment by fgvieira** [comment]
Even though all reads are the same length, there is still data, so I'd expect it to be on the `json`. But not a big issues, if the other plots are there. 😄 
However, these `-1` plots do not show up on `MegaQC`; do you think it could be because of the name? Do you think by using the `anchor` as prefix it will show up?

**Comment by vladsavelyev** [comment]
Yeah, agree that it makes sense to still save the data into JSON. I'll take a look!

Regarding MegaQC, good question - need to look into it.

**Comment by vladsavelyev** [comment]
In a few weeks perhaps, yes. But I doubt that draft PR changes anything with regards to MegaQC
